File: examples/experimental/java-parallel-helloworld/src/main/java/parallel/Serving.java
package org.apache.predictionio.examples.java.parallel;

import org.apache.predictionio.controller.java.EmptyParams;
import org.apache.predictionio.controller.java.LJavaServing;

public class Serving extends LJavaServing<EmptyParams, Query, Float> {

  public Serving() {

  }

  @Override
  public Float serve(Query query, Iterable<Float> predictions) {
    return predictions.iterator().next();
  }

}

File: examples/experimental/scala-parallel-trim-app/src/main/scala/Serving.scala
package org.apache.predictionio.examples.experimental.trimapp

import org.apache.predictionio.controller.LServing

class Serving
  extends LServing[Query, PredictedResult] {

  override
  def serve(query: Query,
    predictedResults: Seq[PredictedResult]): PredictedResult = {
    predictedResults.head
  }
}
